detect.go

```diff
@@ -0,0 +1,66 @@
+package require
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/paketo-buildpacks/nginx"
+	"github.com/paketo-buildpacks/packit/v2"
+	"github.com/paketo-buildpacks/packit/v2/scribe"
+)
+
+const indexFile = "index.html"
+
+type BuildPlanMetadata struct {
+	Version       string `toml:"version,omitempty"`
+	VersionSource string `toml:"version-source,omitempty"`
+	Launch        bool   `toml:"launch"`
+}
+
+func Detect(logger scribe.Emitter) packit.DetectFunc {
+	return func(context packit.DetectContext) (packit.DetectResult, error) {
+		if _, err := WebRoot(context.WorkingDir); err != nil {
+			return packit.DetectResult{}, err
+		}
+
+		result := packit.DetectResult{
+			Plan: packit.BuildPlan{
+				Requires: []packit.BuildPlanRequirement{
+					{
+						Name:     nginx.NGINX,
+						Metadata: BuildPlanMetadata{Launch: true},
+					},
+				},
+			},
+		}
+
+		return result, nil
+	}
+}
+
+func WebRoot(workingDir string) (string, error) {
+	paths := []string{"./", "./public"}
+	publicDir := ""
+	for _, path := range paths {
+		f, err := os.Stat(filepath.Join(workingDir, path, indexFile))
+		if err != nil {
+			if errors.Is(err, os.ErrNotExist) {
+				continue
+			}
+			return "", packit.Fail.WithMessage("failed to stat: %w", err)
+		}
+		if f.IsDir() {
```

Comment:
not sure, but if there is directory `index.html` in the root, but there is an additional file `public.index.html` then we would not find it, right?

Reply:
yes, but should we? Would be quite a strange pattern to put your file in a directory named `index.html` :smile:

Reply:
Well it wouldn't be. There would be a valid file in `public/index.html`, but additionally there is a directory called `index.html` in the root (maybe a mistake?).

Reply:
ah that's what you mean. I guess we could just continue here instead of erroring.

Reply:
I changed it to continue here.